File: cpp_codes/2023/december/18-Game of XOR.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
    int gameOfXor(int N , int A[]) {
        
    int result = 0;

    for (int i = 0; i < N; ++i) {
        int freq = (i + 1) * (N - i);
        
        // If frequency is even, XOR is 0.
        if (freq % 2 == 0) {
            result ^= 0;
        } 
        // If frequency is odd, XOR with the element.
        else {
            result ^= A[i];
        }
    }

    return result;
    }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        
        cin>>N;
        int A[N];
        for(int i=0 ; i<N ; i++)
            cin>>A[i];

        Solution ob;
        cout << ob.gameOfXor(N,A) << endl;
    }
    return 0;
}
// } Driver Code Ends

File: java_codes/2023/december/18-Game Of XOR.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int N = Integer.parseInt(read.readLine());
            
            String S[] = read.readLine().split(" ");
            int[] A = new int[N];
            
            for(int i=0 ; i<N ; i++)
                A[i] = Integer.parseInt(S[i]);

            Solution ob = new Solution();
            System.out.println(ob.gameOfXor(N,A));
        }
    }
}
// } Driver Code Ends

//User function Template for Java
class Solution {
    static int gameOfXor(int N , int[] A) {
        //if size of the array is even, then all the elements occur even times in all subarrays
        //Hence, their xor gives 0
        if(N % 2 == 0){
            return 0;
        }
        
        //if size of the array is odd, then elements at odd index occurs odd times,
        //elements at even indices occurs even times therefore, only taking xor of
        //elements at odd indices
        int ans = 0;
        for(int i = 0; i<N; i+=2){
            ans ^= A[i];
        }
        
        return ans;
    }
};